### ISSUE TITLE
Machines stuck Provisioning, Node.Spec.ProviderID not set

### ISSUE THREAD

**Issue by wcurry** [kubernetes-sigs/cluster-api-provider-vsphere#1107]
/kind bug

**What steps did you take and what happened:**

We have seen this bug following testing deletion of VMs from VSphere as well as deletion of Machine objects in CAPI. It has been infrequent, and hard to reproduce, but in the last two days we've hit it twice and spent some time looking more closely into it.

The Machine and VSphereMachine objects have the providerID, but it is not set in the Node spec.

Setting the providerID manually by editing the Node resource causes the Machine to finish provisioning.

**What did you expect to happen:**

Machine provisions.

**Anything else you would like to add:**

## capi-controller-mgr-logs.txt
```
f40c@M-C02V11NZHTDF:cloud-provider on heads/v1.17.11 (hydrogen)$ kubectl -n k8s-capi-ci-2209951 logs capi-controller-manager-577485b47f-8z67s manager |grep onprem-2209951-worker-5d457f4fdc-snv9c|tail -40

E0205 21:19:55.524989       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:19:57.490246       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:23:49.132929       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:23:49.132939       1 machinehealthcheck_controller.go:480] controllers/MachineHealthCheck "msg"="Unable to retrieve machine from node" "error"="expecting one machine for node onprem-2209951-worker-5d457f4fdc-snv9c, got []"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:23:49.132970       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:23:49.133005       1 machinehealthcheck_controller.go:480] controllers/MachineHealthCheck "msg"="Unable to retrieve machine from node" "error"="expecting one machine for node onprem-2209951-worker-5d457f4fdc-snv9c, got []"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:23:50.046893       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:24:14.156980       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:24:14.157735       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:24:14.159238       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:24:14.161651       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:24:17.326660       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:24:58.501632       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:26:10.613845       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:26:10.617422       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:26:10.623659       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:27:32.071711       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:28:49.244430       1 machinehealthcheck_controller.go:480] controllers/MachineHealthCheck "msg"="Unable to retrieve machine from node" "error"="expecting one machine for node onprem-2209951-worker-5d457f4fdc-snv9c, got []"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:28:49.244491       1 machinehealthcheck_controller.go:480] controllers/MachineHealthCheck "msg"="Unable to retrieve machine from node" "error"="expecting one machine for node onprem-2209951-worker-5d457f4fdc-snv9c, got []"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:28:49.244535       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:28:49.244641       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:28:50.138922       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:29:04.863233       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:29:17.436990       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:29:51.989893       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:30:14.879386       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:30:14.887475       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:30:14.894233       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:30:14.896562       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:29.125848       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:29.128897       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:29.133639       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:49.351048       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:49.351139       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:49.351049       1 machinehealthcheck_controller.go:480] controllers/MachineHealthCheck "msg"="Unable to retrieve machine from node" "error"="expecting one machine for node onprem-2209951-worker-5d457f4fdc-snv9c, got []"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:49.351179       1 machinehealthcheck_controller.go:480] controllers/MachineHealthCheck "msg"="Unable to retrieve machine from node" "error"="expecting one machine for node onprem-2209951-worker-5d457f4fdc-snv9c, got []"  "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:33:50.247867       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:34:17.548816       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:34:52.987240       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
E0205 21:34:53.772238       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="onprem-2209951-worker-5d457f4fdc-snv9c"
```
## get-machine.txt
```
f40c@M-C02V11NZHTDF:cloud-provider on heads/v1.17.11 (hydrogen)$ kubectl -n k8s-capi-ci-2209951 get machines onprem-2209951-worker-5d457f4fdc-snv9c -oyaml
apiVersion: cluster.x-k8s.io/v1alpha3
kind: Machine
metadata:
  creationTimestamp: "2021-02-05T18:57:03Z"
  finalizers:
  - machine.cluster.x-k8s.io
  generateName: onprem-2209951-worker-5d457f4fdc-
  generation: 3
  labels:
    cluster.x-k8s.io/cluster-name: onprem-2209951
    cluster.x-k8s.io/deployment-name: onprem-2209951-worker
    machine-template-hash: "1801390987"
    worker-node: "true"
  name: onprem-2209951-worker-5d457f4fdc-snv9c
  namespace: k8s-capi-ci-2209951
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1alpha3
    blockOwnerDeletion: true
    controller: true
    kind: MachineSet
    name: onprem-2209951-worker-5d457f4fdc
    uid: bfb83db5-2f14-4395-b4eb-00ce67179b41
  resourceVersion: "8752335765"
  selfLink: /apis/cluster.x-k8s.io/v1alpha3/namespaces/k8s-capi-ci-2209951/machines/onprem-2209951-worker-5d457f4fdc-snv9c
  uid: 7a2a249e-5b01-475a-8614-e0467424f60e
spec:
  bootstrap:
    configRef:
      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
      kind: KubeadmConfig
      name: onprem-2209951-worker-5wvz6
      namespace: k8s-capi-ci-2209951
      uid: 046893e8-5d13-4c6a-a9b6-e3c2a1c898ed
    dataSecretName: onprem-2209951-worker-5wvz6
  clusterName: onprem-2209951
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
    kind: VSphereMachine
    name: onprem-2209951-ea61263f681a8f6fe0315d7ad12f6953-bgp4f
    namespace: k8s-capi-ci-2209951
    uid: c9801f9a-f090-4ce9-9517-92a1dcfd5d1e
  providerID: vsphere://42214357-e2d8-a4f7-55b9-04f2a118a824
  version: v1.17.11
status:
  addresses:
  - address: 10.17.40.88
    type: ExternalIP
  bootstrapReady: true
  conditions:
  - lastTransitionTime: "2021-02-05T18:57:49Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2021-02-05T18:57:03Z"
    status: "True"
    type: BootstrapReady
  - lastTransitionTime: "2021-02-05T18:57:49Z"
    status: "True"
    type: InfrastructureReady
  - lastTransitionTime: "2021-02-05T18:57:48Z"
    reason: NodeProvisioning
    severity: Warning
    status: "False"
    type: NodeHealthy
  infrastructureReady: true
  lastUpdated: "2021-02-05T18:57:03Z"
  observedGeneration: 3
  phase: Provisioning
```
## get-machines-o-wide.txt
```
f40c@M-C02V11NZHTDF:cloud-provider on heads/v1.17.11 (hydrogen)$ kubectl -n k8s-capi-ci-2209951 get machines  -owide
NAME                                     PROVIDERID                                       PHASE          VERSION    NODENAME
onprem-2209951-645rm                     vsphere://42213fda-2e10-87bf-376d-b57117f1fc27   Running        v1.17.11   onprem-2209951-645rm
onprem-2209951-q9pjf                     vsphere://4221f51c-8028-e523-0347-b7ca45de7453   Running        v1.17.11   onprem-2209951-q9pjf
onprem-2209951-worker-5d457f4fdc-csq5s   vsphere://42213f27-1e8a-cae7-3bb2-5ad8217137cd   Running        v1.17.11   onprem-2209951-worker-5d457f4fdc-csq5s
onprem-2209951-worker-5d457f4fdc-lbcd9   vsphere://4221a491-511c-793e-0e4c-121da82ad512   Running        v1.17.11   onprem-2209951-worker-5d457f4fdc-lbcd9
onprem-2209951-worker-5d457f4fdc-snv9c   vsphere://42214357-e2d8-a4f7-55b9-04f2a118a824   Provisioning   v1.17.11
onprem-2209951-x2bb9                     vsphere://42212b82-14a9-24d6-a092-861be2f1d79d   Running        v1.17.11   onprem-2209951-x2bb9
```
## get-node.txt
```
f40c@M-C02V11NZHTDF:cloud-provider on heads/v1.17.11 (hydrogen)$ kubectl --kubeconfig /Users/f40c/.kube/onprem-2209951.kubeconfig -n monitoring get nodes onprem-2209951-worker-5d457f4fdc-snv9c -oyaml
apiVersion: v1
kind: Node
metadata:
  annotations:
    kubeadm.alpha.kubernetes.io/cri-socket: /var/run/containerd/containerd.sock
    node.alpha.kubernetes.io/ttl: "15"
    volumes.kubernetes.io/controller-managed-attach-detach: "true"
  creationTimestamp: "2021-02-05T18:57:44Z"
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: onprem-2209951-worker-5d457f4fdc-snv9c
    kubernetes.io/os: linux
  name: onprem-2209951-worker-5d457f4fdc-snv9c
  resourceVersion: "537872"
  selfLink: /api/v1/nodes/onprem-2209951-worker-5d457f4fdc-snv9c
  uid: 4d0919f9-4525-48d3-91c3-59db3db70c6f
spec:
  taints:
  - effect: NoSchedule
    key: node.cloudprovider.kubernetes.io/uninitialized
    value: "true"
status:
  addresses:
  - address: 10.17.40.88
    type: InternalIP
  - address: onprem-2209951-worker-5d457f4fdc-snv9c
    type: Hostname
  allocatable:
    cpu: 6776m
    ephemeral-storage: "23658268224"
    hugepages-2Mi: "0"
    memory: 20504400Ki
    pods: "110"
  capacity:
    cpu: "8"
    ephemeral-storage: 25670864Ki
    hugepages-2Mi: "0"
    memory: 32939856Ki
    pods: "110"
  conditions:
  - lastHeartbeatTime: "2021-02-05T21:38:49Z"
    lastTransitionTime: "2021-02-05T18:57:46Z"
    message: kubelet has sufficient memory available
    reason: KubeletHasSufficientMemory
    status: "False"
    type: MemoryPressure
  - lastHeartbeatTime: "2021-02-05T21:38:49Z"
    lastTransitionTime: "2021-02-05T18:57:46Z"
    message: kubelet has no disk pressure
    reason: KubeletHasNoDiskPressure
    status: "False"
    type: DiskPressure
  - lastHeartbeatTime: "2021-02-05T21:38:49Z"
    lastTransitionTime: "2021-02-05T18:57:46Z"
    message: kubelet has sufficient PID available
    reason: KubeletHasSufficientPID
    status: "False"
    type: PIDPressure
  - lastHeartbeatTime: "2021-02-05T21:38:49Z"
    lastTransitionTime: "2021-02-05T18:58:26Z"
    message: kubelet is posting ready status. AppArmor enabled
    reason: KubeletReady
    status: "True"
    type: Ready
  daemonEndpoints:
    kubeletEndpoint:
      Port: 10250
  images:
  - names:
    - artifactory/docker/app00738/newrelic/infrastructure-k8s@sha256:ecc5a69cef1c1038fafd7ffa73d189be2857840aa98f7bd835dda1f466e635ae
    sizeBytes: 182920632
  - names:
    - k8s.gcr.io/etcd@sha256:4afb99b4690b418ffc2ceb67e1a17376457e441c1f09ab55447f0aaf992fa646
    - k8s.gcr.io/etcd:3.4.3-0
    sizeBytes: 100947667
  - names:
    - gitlab-registry/sre/public/newrelic-gostatsd@sha256:edbd7677875c0d0557a95c93ebbccd677fb7e54434664b0c15ebe81a9c7f8e27
    - gitlab-registry/sre/public/newrelic-gostatsd:1570649-696c2ee3
    sizeBytes: 61676359
  - names:
    - k8s.gcr.io/kube-apiserver@sha256:a4441e7db659373f6ffc3a3795c2f5e23c8375b39293f4d322f62d094af093e2
    - k8s.gcr.io/kube-apiserver:v1.17.11
    sizeBytes: 50683403
  - names:
    - artifactory/docker/app00738/kube-proxy@sha256:f80ff6fe6305229e52b9052d9b6f4787c7b61dcf8329309ef6504f581c54789d
    - k8s.gcr.io/kube-proxy@sha256:087bea5dd606a48d744d85c794c9bc138fe5c03415e6932a3bdbb4322e47f6f4
    - artifactory/docker/app00738/kube-proxy:v1.17.11
    - k8s.gcr.io/kube-proxy:v1.17.11
    sizeBytes: 49238132
  - names:
    - k8s.gcr.io/kube-controller-manager@sha256:1c7dec58d055a14f563266a83190590d1c2e9d88197c8675d73612228547a34a
    - k8s.gcr.io/kube-controller-manager:v1.17.11
    sizeBytes: 48840799
  - names:
    - k8s.gcr.io/kube-scheduler@sha256:a45a168c6e444f704219372f029cfc86577db67e82c3828c504f24375a117d35
    - k8s.gcr.io/kube-scheduler:v1.17.11
    sizeBytes: 33828316
  - names:
    - artifactory/docker/app00738/bash-image@sha256:5d6dad79c989cb0f8d0a7ce985c0f52b554b03b1bc7960a93b2036b49e87eade
    sizeBytes: 23846881
  - names:
    - artifactory/docker/app00738/coreos/flannel@sha256:34860ea294a018d392e61936f19a7862d5e92039d196cac9176da14b2bbd0fe3
    - artifactory/docker/app00738/coreos/flannel:v0.13.0-amd64
    sizeBytes: 19386553
  - names:
    - artifactory/docker/app00738/conntrack-stats-exporter@sha256:3ddd8402676c514a610071f094332fa4ff820836281951fdc8144c6cb8ffb5a5
    sizeBytes: 14206692
  - names:
    - k8s.gcr.io/coredns@sha256:7ec975f167d815311a7136c32e70735f0d00b73781365df1befd46ed35bd4fe7
    - k8s.gcr.io/coredns:1.6.5
    sizeBytes: 13239960
  - names:
    - artifactory/docker/app00738/prom/node-exporter@sha256:3b6a80b7cabd0bae3c3ffa903387e1a659aedeecefe30b97805324e179cbe710
    - artifactory/docker/app00738/prom/node-exporter:v1.0.0-rc.0
    sizeBytes: 11966200
  - names:
    - k8s.gcr.io/pause@sha256:f78411e19d84a252e53bff71a4407a5686c46983a2c2eeed83929b888179acea
    - k8s.gcr.io/pause:3.1
    sizeBytes: 317164
  nodeInfo:
    architecture: amd64
    bootID: f6e1b842-0f33-41eb-bc2f-0a17f09acb10
    containerRuntimeVersion: containerd://1.3.4
    kernelVersion: 4.15.0-112-generic
    kubeProxyVersion: v1.17.11
    kubeletVersion: v1.17.11
    machineID: 43d4e164e6db49d79e48f347c0209c5b
    operatingSystem: linux
    osImage: Ubuntu 18.04.5 LTS
    systemUUID: 57432142-D8E2-F7A4-55B9-04F2A118A824

```
## get-vspheremachine.txt
```
f40c@M-C02V11NZHTDF:cloud-provider on heads/v1.17.11 (hydrogen)$ kubectl -n k8s-capi-ci-2209951 get vspheremachine onprem-2209951-ea61263f681a8f6fe0315d7ad12f6953-bgp4f -oyaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: VSphereMachine
metadata:
  annotations:
    cluster.x-k8s.io/cloned-from-groupkind: VSphereMachineTemplate.infrastructure.cluster.x-k8s.io
    cluster.x-k8s.io/cloned-from-name: onprem-2209951-ea61263f681a8f6fe0315d7ad12f6953
  creationTimestamp: "2021-02-05T18:57:03Z"
  finalizers:
  - vspheremachine.infrastructure.cluster.x-k8s.io
  generation: 2
  labels:
    cluster.x-k8s.io/cluster-name: onprem-2209951
    cluster.x-k8s.io/deployment-name: onprem-2209951-worker
    machine-template-hash: "1801390987"
    worker-node: "true"
  name: onprem-2209951-ea61263f681a8f6fe0315d7ad12f6953-bgp4f
  namespace: k8s-capi-ci-2209951
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1alpha3
    blockOwnerDeletion: true
    controller: true
    kind: Machine
    name: onprem-2209951-worker-5d457f4fdc-snv9c
    uid: 7a2a249e-5b01-475a-8614-e0467424f60e
  resourceVersion: "8752335763"
  selfLink: /apis/infrastructure.cluster.x-k8s.io/v1alpha3/namespaces/k8s-capi-ci-2209951/vspheremachines/onprem-2209951-ea61263f681a8f6fe0315d7ad12f6953-bgp4f
  uid: c9801f9a-f090-4ce9-9517-92a1dcfd5d1e
spec:
  cloneMode: linkedClone
  datacenter: /DC
  datastore: /DC/datastore/cl...
  diskGiB: 25
  folder: /DC/vm/dev
  memoryMiB: 32768
  network:
    devices:
    - dhcp4: true
      networkName: vlanname
  numCPUs: 8
  providerID: vsphere://42214357-e2d8-a4f7-55b9-04f2a118a824
  resourcePool: /Denver/host/cl...
  server: vcsa.domain.net
  template: /Denver/vm/dev/Templates/ubuntu-1804-kube-v1.17.11
status:
  addresses:
  - address: 10.17.40.88
    type: ExternalIP
  conditions:
  - lastTransitionTime: "2021-02-05T18:57:49Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2021-02-05T18:57:49Z"
    status: "True"
    type: VMProvisioned
  network:
  - connected: true
    ipAddrs:
    - 10.17.40.88
    macAddr: 00:50:16:a1:7c:df
    networkName: vlanname
  ready: true


```
## logs-capv-controller-grep-snv9c-tail-20.txt
```
f40c@M-C02V11NZHTDF:cloud-provider on heads/v1.17.11 (hydrogen)$ kubectl -n k8s-capi-ci-2209951 logs capv-controller-manager-54bf669864-djppp manager |grep snv9c|tail -20
I0205 21:07:20.940032       1 vspherevm_controller.go:248] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="resource patch was not required"  "local-resource-version"="8752335753" "remote-resource-version"="8752335753"
I0205 21:07:20.940081       1 controller.go:272] controller-runtime/controller "msg"="Successfully Reconciled" "controller"="vspherevm" "name"="onprem-2209951-worker-5d457f4fdc-snv9c" "namespace"="k8s-capi-ci-2209951"
I0205 21:17:15.881522       1 util.go:68] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="vm found by bios uuid"  "vmref"={"Type":"VirtualMachine","Value":"vm-33152"}
I0205 21:17:16.047200       1 service.go:269] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="powered on"
I0205 21:17:16.047222       1 vspherevm_controller.go:339] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="vm bios-uuid"  "biosuuid"="42214357-e2d8-a4f7-55b9-04f2a118a824"
I0205 21:17:16.047224       1 vspherevm_controller.go:359] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="VSphereVM is ready"
I0205 21:17:16.048304       1 vspherevm_controller.go:248] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="resource patch was not required"  "local-resource-version"="8752335753" "remote-resource-version"="8752335753"
I0205 21:17:16.048356       1 controller.go:272] controller-runtime/controller "msg"="Successfully Reconciled" "controller"="vspherevm" "name"="onprem-2209951-worker-5d457f4fdc-snv9c" "namespace"="k8s-capi-ci-2209951"
I0205 21:27:10.785081       1 util.go:68] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="vm found by bios uuid"  "vmref"={"Type":"VirtualMachine","Value":"vm-33152"}
I0205 21:27:10.976945       1 service.go:269] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="powered on"
I0205 21:27:10.976983       1 vspherevm_controller.go:339] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="vm bios-uuid"  "biosuuid"="42214357-e2d8-a4f7-55b9-04f2a118a824"
I0205 21:27:10.977023       1 vspherevm_controller.go:359] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="VSphereVM is ready"
I0205 21:27:10.978018       1 vspherevm_controller.go:248] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="resource patch was not required"  "local-resource-version"="8752335753" "remote-resource-version"="8752335753"
I0205 21:27:10.978056       1 controller.go:272] controller-runtime/controller "msg"="Successfully Reconciled" "controller"="vspherevm" "name"="onprem-2209951-worker-5d457f4fdc-snv9c" "namespace"="k8s-capi-ci-2209951"
I0205 21:37:05.936721       1 util.go:68] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="vm found by bios uuid"  "vmref"={"Type":"VirtualMachine","Value":"vm-33152"}
I0205 21:37:06.115956       1 service.go:269] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="powered on"
I0205 21:37:06.116100       1 vspherevm_controller.go:339] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="vm bios-uuid"  "biosuuid"="42214357-e2d8-a4f7-55b9-04f2a118a824"
I0205 21:37:06.116109       1 vspherevm_controller.go:359] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="VSphereVM is ready"
I0205 21:37:06.117120       1 vspherevm_controller.go:248] capv-controller-manager/vspherevm-controller/k8s-capi-ci-2209951/onprem-2209951-worker-5d457f4fdc-snv9c "msg"="resource patch was not required"  "local-resource-version"="8752335753" "remote-resource-version"="8752335753"
I0205 21:37:06.117155       1 controller.go:272] controller-runtime/controller "msg"="Successfully Reconciled" "controller"="vspherevm" "name"="onprem-2209951-worker-5d457f4fdc-snv9c" "namespace"="k8s-capi-ci-2209951"

```

**Environment:**

- Cluster-api-provider-vsphere version:  v0.7.1
- Kubernetes version: (use `kubectl version`): v1.17.11
- OS (e.g. from `/etc/os-release`): ubuntu-1804-kube-v1.17.11 OVA

Thank you!

**Comment by yastij** [comment]
@wcurry - can you share which version of the CPI you're using + logs from the cloud-controller-manager ?

**Comment by rwlib3** [comment]
experiencing the same thing. Nodes are deploying (visible via kubectl get machines and get vspsheremachines), but stuck in "Provisioning"...in the management cluster.

In the k8s cluster the nodes are visible and show "Ready" but no External IP is visible in the k8s cluster side.

capi-controller-manager log
E0218 03:41:53.126751       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="ts-kubeflow-sac-prod-md1-6ffd74bd5b-khxzr"
E0218 03:42:09.155577       1 machine_controller.go:685] controllers/Machine "msg"="Unable to retrieve machine from node" "error"="no matching Machine"  "node"="ts-kubeflow-sac-prod-md1-6ffd74bd5b-khxzr"

**Comment by wcurry** [comment]
> @wcurry - can you share which version of the CPI you're using + logs from the cloud-controller-manager ?

CPI v1.2.0. The logs are long gone. I'll try to get them next time.

**Comment by MaxRink** [comment]
> > @wcurry - can you share which version of the CPI you're using + logs from the cloud-controller-manager ?
> 
> CPI v1.2.0. The logs are long gone. I'll try to get them next time.

Yeah, update that to at least 1.2.1
1.2.0 has serious bugs

**Comment by wcurry** [comment]
I hit this again while working on a Kubernetes version bump that we had to scrap due to both KCP and MD rollouts getting stuck. The MD rollout was stuck with this issue. I bumped CAPV to 0.7.2, CPI to 1.2.1 and all the other dependencies to what's in the repo, and still hit the issue.

We're going to wait until 0.7.7 before looking at this again as we hit issues with 0.7.6: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/1150
